lib/clients/public.js

```diff
@@ -1,8 +1,9 @@
 const request = require('request');
 const { Readable } = require('stream');
+const DEFAULT_TIMEOUT = 10 * 1000; // 10 sec
 
 class PublicClient {
-  constructor(apiURI = 'https://api.gdax.com') {
+  constructor(apiURI = 'https://api.gdax.com', options = {}) {
```

Comment:
Maybe this is personal preference, but I'd prefer to see the defaults for all options set in the method signature:

```js
constructor(apiURI = 'https://api.gdax.com', { timeout = DEFAULT_TIMEOUT } = {})

Reply:
I generally agree, but there's some additional input checking in the constructor in this instance.

Reply:
Good catch on the authclient constructor

Reply:
> I generally agree, but there's some additional input checking in the constructor in this instance.

 But this input checking is mostly unnecessary and actually invalid. A `timeout` of `0` or `undefined` is perfectly valid using `request` library, and any other validation of `timeout` is already handled by the `request` library. e.g. negative values are already normalized.